RELEASE_NOTES.md: Add initial release notes for v0.3.0

oyvindronningstad opened the pull request:
Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

hakonfam commented:
I'm not fully sold on the git SHA, since you can always grep for the commit message to find the commit, but then its not evident to the reader that the entries in this file is actually commit messages. This is just a nit, and feel free to ignore, but could we place them at the end of the line instead of at the start?

kyberdin reviewed (changes requested):
I tend to agree with @hakonfam here. AFAICT this doesn't add much more than what is already given in the git history. It's great that the commit messages are so self-explanatory, but all of this information is already available in a diff between version tags or SHAs.

I'd suggest rewording these release notes to be a higher level description of the notable changes. For example (some random entries):

 ```markdown
 * Better support for hexadecimal/octal/binary numbers in CDDL
   * E.g. in size ranges or value limits
 * Added shorthand option for `--default-max-qty` and expanded documentation
 * Improved readability and documentation for CBOR
 * Changed command line entry point from `cddl_gen` to `cddl-gen`
 * CMake files are now generated alongside generated code, which can be used when building your app.
   * See `README.md` for details
 * Tests are now run with twister instead of custom bash scripts
 * Various test improvements for `manifest14`
 ...etc.
 ```
Each commit doesn't need its own entry per se, just important points.

Thoughts?

oyvindronningstad commented:
@kyberdin Good points. I redid it significantly now.

kyberdin commented on RELEASE_NOTES.md:
```diff
@@ -0,0 +1,45 @@
+### CDDL language
+ * Add more support for hexadecimal/octal/binary numbers in CDDL. E.g. in size ranges or value limits
+ * Add support for ranges with three dots in CDDL files
+ * Change CDDL quantifier syntax from x**y to x*y
```
Since this is markdown file anyway, I'd suggest using the inline "\`" marker for these lines, since markdown will take asterisks as either bold or italics, like it did here in the render between with "x**y to x*y" instead of `x**y` to `x*y`.

kyberdin commented on RELEASE_NOTES.md:
```diff
@@ -0,0 +1,45 @@
+ * Add support for the .bits operator.
+
+### Command line
+ * Change command line name from cddl_gen to cddl-gen
+ * Add --dq as a shorthand for --default-max-qty
```
Would also suggest inline markdown syntax here for `cddl_gen`, `cddl-gen`, `--dq`, and `--default-max-qty`.

Same goes for other code bits like function calls in the file, like `target_cddl_source()`. File names can also look clearer sometimes, but no strong opinion on that here.

kyberdin commented on RELEASE_NOTES.md:
```diff
@@ -0,0 +1,45 @@
+ * cbor_encode.c: Fix get_result_len and add big-endian support
+ * cddl_gen.py: Fix bug where range checks were dropped
+ * cbor_decode.c: Fix bug where a 'nil' could be interpreted as true
+ * cddl_gen.py: Fix parsing of ranges with no max (n* or n**)
```
Similar to the above about inline for asterisks to avoid markdown syntax converting them unintentionally.

kyberdin reviewed (changes requested):
The changes from the last iteration look fantastic, just a few nits about markdown syntax.

oyvindronningstad commented:
Thanks @kyberdin I added quite a few backticks now :).

I also added a list of unsupported CDDL features as suggested in https://github.com/NordicSemiconductor/cddl-gen/issues/126

kyberdin commented on RELEASE_NOTES.md:
```diff
@@ -0,0 +1,67 @@
+
+## Unsupported CDDL features
+
+Not all features outlined in the CDDL spec (https://datatracker.ietf.org/doc/html/rfc8610) are supported by cddl-gen.
```
Markdown can support rendering this as a hyperlink:
```suggestion
Not all features outlined in the [CDDL spec](https://datatracker.ietf.org/doc/html/rfc8610) are supported by cddl-gen.
```
